docs/developers/components/reference.md

```diff
@@ -97,21 +97,511 @@ If a `config.yaml` is defined, it will **not** be merged with the default config
+export function handleComponent(scope) {
+  const staticFiles = new Map();
+
+  scope.options.on('change', (key, value, config) => {
+    if (key[0] === 'files' || key[0] === 'urlPath') {
+      // If the files or urlPath options change, we need to reinitialize the static files map
+      staticFiles.clear();
+      logger.info(`Static files reinitialized due to change in ${key.join('.')}`);
+    }
+  });
+
+  scope.handleEntry((entry) => {
+    if (entry.entryType === 'directory') {
+      logger.info(`Cannot serve directories. Update the files option to only match files.`);
+      return;
+    }
+
+    switch (entry.eventType) {
+      case 'add':
+      case 'change':
+        // Store / Update the file contents in memory for serving
+        staticFiles.set(entry.urlPath, entry.contents);
+        break;
+      case 'unlink':
+        // Remove the file from memory when it is deleted
+        staticFiles.delete(entry.urlPath);
+        break;
+    }
+  });
+
+  scope.server.http((req, next) => {
+    if (req.method !== 'GET') return next(req);
+
+    // Attempt to retrieve the requested static file from memory
+    const staticFile = staticFiles.get(req.pathname);
+
+    return staticFile ? {
+      statusCode: 200,
+      body: staticFile,
+    } : {
+      statusCode: 404,
+      body: 'File not found',
+    }
+  }, { runFirst: true });
+}
+```
+
+In this example, the entry handler method passed to `handleEntry` will manage the map of static files in memory using their computed `urlPath` and the `contents`. If the config file changes (and thus a new default file or url path is specified) the extension will clear the file map as well to remove artifacts. Furthermore, it uses the `server.http()` middleware to hook into the HTTP request handling.
+
+This example is heavily simplified, but it demonstrates how the different key parts of `scope` can be used together to provide a performant and reactive application experience.
+
+### `handleComponent(scope: Scope): void | Promise<void>`
+
+Parameters:
+
+- **scope** - [`Scope`](#class-scope) - An instance of the `Scope` class that provides access to the component's configuration, resources, and other APIs.
+
+Returns: `void | Promise<void>`
+
+This is the only method an extension module must export. It can be async and is awaited by the component loader. The `scope` argument provides access to the component's configuration, resources, and other APIs.
+
+### Class: `Scope`
+
+- Extends [`EventEmitter`](https://nodejs.org/docs/latest/api/events.html#class-eventemitter)
+
+#### Event: `'close'`
+
+Emitted after the scope is closed via the `close()` method. The scope is not 
+
+#### Event: `'error'`
+
+- **error** - `unknown` - The error that occurred.
+
+#### Event: `'ready'`
+
+Emitted when the Scope is ready to be used after loading the associated config file. It is awaited by the component loader, so it is not necessary to await it within the `handleComponent()` method.
+
+#### `scope.ready()`
```

Comment:
Should this be documented? (Is there meaningful usage for users of this API?)

Reply:
Its public - so like if someone inspected the `scope` they'd see it on there. I prefer to document any public methods.

Reply:
done